ggml.h

```diff
@@ -1493,7 +1493,8 @@ extern "C" {
             int                  p1,
             int                  d0,
             int                  d1,
-            bool                 is_2D);
+            bool                 is_2D,
+            enum ggml_type       dst_type);
```

Comment:
I'm not sure about this API.

Long term we would like all ops to be able to output certain type which would mean we would have to extend all ops with `dst_type`. Is this OK?

We already started using another pattern with `ggml_mul_mat_set_prec()`.
Not that it is great, but we might want to look for consistency here.

Somewhat related: https://github.com/ggerganov/ggml/issues/455

Reply:
I think this is something that we will need to deal with when we add support for full F16 pipeline. I don't think adding a `dst_type` parameter to every op would be a good way to do this, a setting in `ggml_context` that sets the output type for all the ops it would make more sense, but as it is now, I don't think we have a better way to handle this.

Reply:
For now should we merge it like this and figure out later?

Reply:
I think so.